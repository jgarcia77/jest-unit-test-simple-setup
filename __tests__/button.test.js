import React from 'react';
import { shallow } from 'enzyme';
import Button from '../button';
import renderer from 'react-test-renderer';

describe("When component is mounted", () => {
    it("Then it should contain a button", () => {
        const wrapper = shallow(<Button text="Click Me !" onClick={() => {}} />);
        const button = wrapper.find('button');

        expect(button.exists()).toEqual(true);
    });

    it("And it should match snapshot", () => {
        const json = renderer.create(<Button text="Click Me !" onClick={() => {}} />);
        
        expect(json).toMatchSnapshot();
    });
});