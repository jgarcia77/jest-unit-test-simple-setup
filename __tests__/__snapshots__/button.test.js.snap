// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When component is mounted And it should match snapshot 1`] = `
<button
  onClick={[Function]}
>
  Click Me !
</button>
`;
